### PR TITLE
fix: auth header error msg 

### PR DESCRIPTION
**Pull request by mehulmathur001**
**Summary:**  
/claim #1665

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by mehulmathur001** [comment]
@tusharmath @amitksingh1490 can you please review this PR ?

**Comment by amitksingh1490** [comment]
error messages are correctly fixed but the actual problem is in basic.rs
```async fn verify(&self, req_ctx: &RequestContext) -> Verification {
    // if the decode fails below line returns NONE. Which should not happen as the headers were passed but actually they were invalid
    let header = req_ctx.allowed_headers.typed_get::<Authorization<Basic>>();
    println!("{:?}", header);
    let Some(header) = header else {
        return Verification::fail(Error::Missing);
    };

    if self.verifier.check(header.username(), header.password()) {
        Verification::succeed()
    } else {
        Verification::fail(Error::Invalid)
    }
}
```

**Comment by mehulmathur001** [comment]
@amitksingh1490 made some changes. Please review.

File: src/auth/basic.rs [hunk 1]
```diff
@@ -18,14 +18,16 @@ impl Verify for BasicVerifier {
...
+            if self.verifier.check(header.username(), header.password()) {
+                Verification::succeed()
+            } else {
+                Verification::fail(Error::Invalid)
+            }
+        } else if req_ctx.allowed_headers.contains_key("authorization") {
+            return Verification::fail(Error::Invalid);
         } else {
-            Verification::fail(Error::Invalid)
+            return Verification::fail(Error::Missing);
```
Comment (ssddOnTop) [inline src/auth/basic.rs:30]: Do not use `contains_key` to check for the header, it adds unnecessary cost to the code. the problem is with de-serialization, check #1673 (feel free to use the code)

**Review by ssddOnTop — APPROVED** [review]

File: src/auth/basic.rs [hunk 2]
```diff
@@ -108,6 +115,18 @@ testuser3:{SHA}Y2fEjdGT1W6nsLqtJbGUVeUp9e4=
...
+    async fn verify_auth_failure() {
+        let provider = setup_provider();
+        let mut req_ctx = RequestContext::default();
+        req_ctx.allowed_headers.insert(
+            "Authorization",
+            HeaderValue::from_static("Basic dGVzdHVzZXIyOm15cGFzc3dvcmQ"),
+        );
+        let validation = provider.verify(&req_ctx).await;
+        assert_eq!(validation, Verification::fail(Error::Invalid));
+    }
```
Comment (tusharmath) [inline src/auth/basic.rs:128]: Add an integration test for this case.

**Comment by amitksingh1490** [comment]
`resolve_token` function in `jwt_verify.rs` should also have similar changes which are done for basic auth. Also add integration test for this scenario
```rust
fn resolve_token(&self, request: &RequestContext) -> Option<String> {
        let value = request.allowed_headers.typed_get::<Authorization<Bearer>>();

        value.map(|token| token.token().to_owned())
    }
```

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**